components/x-teaser-timeline/src/lib/transform.js

```diff
@@ -19,7 +19,11 @@ const groupItemsByLocalisedDate = (
 
 	items.forEach((item, index) => {
 		const localDateTime = getLocalisedISODate(item.publishedDate, timezoneOffset)
-		const localDate = getDateOnly(localDateTime)
+		// N.B. This is to keep a previous hack that existed. Changing this may break existing dates.
+		// https://github.com/Financial-Times/x-dash/commit/b4fcedcd3bcee721a7a03877b9bbcea18fdd4e1c
+		// This date won't represent the timezone. Using date-fns respects the timezone, so it's not used here.
+
+		const localDate = localDateTime.substring(0, 10)
```

Comment:
How come this might break existing date if it's doing the same as before?

before we had this
```
export const getDateOnly = date => date.substr(0, 10);
```

Reply:
Sorry will clarify!

This PR moves the hack to use a substring to get the date to only be used once, in this file. Previously it was used wherever getDateOnly was used.

The reason I've kept it in this file is that removing this changes the snapshot test, and I'm not certain what the desired behaviour should be. If this is SSR, and we're not serving from the users region, then I'd rather not mess with it. If this is client side rendered, then I'm fairly confident with updating it.

Reply:
I've done a [search in Github ](https://github.com/search?q=org%3AFinancial-Times+x-teaser-timeline&type=code) and there are a couple of `x-teaser` clients:

- ft-app - client
- next-myft-page - server

So `next-myft-page` would be affected, isn't it?

Reply:
Seemingly so!

I mistakenly did the same thing, but just with `x-teaser` and found it used on both the client and the server in next-article..

With this change I'll include some notes in the release notes, but do you think it's worth me checking / getting approval from the next-myft-page and the ft-app team?

Reply:
next-myft-page it's us (content-disco) and ft-app that's the App team.

I was thinking, could you test this update to check how's working in `next-myft-page`?

Reply:
Apologies @gyss I'm unable to get the next-myft-page running!

It's showing the 'you need to install next-router' error, despite next-router seemingly being set up correctly (make test works in next-router directory).

Is this just a 'me' issue?

Reply:
that's easy to fix. Just install next-router in local with the same `npm` and `node` versions than `next-myft-page`. Then npm link next-router into next-myft-page

Reply:
Ok so I've got past that issue with help from this thread: https://financialtimes.slack.com/archives/C041V9QA7/p1655480220646879

But that's just leading me to the next issue which is, make .env failing due to vault issues, despite being logged in.

Either way, how necessary is testing this in next-myft-page? It feels that testing middleware in each of it's consumers leads to a cascading level of effort.

Alternatively, are there any tests I could add / fix either here or in next-myft-page that would reassure us that these modules are working as expected?

Reply:
Why would `make .env` have any error? did you run this command before in other projects?

Well, I'm not asking to test every one of the consumers, just one to check the component continues working as expected.

I'm concerned about this PR for some reasons:
- Not sure what's the benefit of this update
- Neither me or my teammates have enough background in this repo to safely approve it. That's why I'd like to make sure by testing
- Deploying a possible issue in the `ft-app` could be problematic since the release system is slower 

Apart from that, looks like there are plenty of snapshots tests in `x-teaser` and `x-teaser-timeline`. But, I was thinking if you could add some unit tests for `getLocalisedISODate()` and `getTitleForItemGroup()` so we can make sure at a lower level that the results are the same

Reply:
Not sure if I've ran `make .env` successfully, haven't needed to AFAIK

I've added a few tests for both those functions, and I ran these tests successfully against the previous version - Please take the horrendous screenshot below for proof!

<img width="1464" alt="image" src="https://github.com/Financial-Times/x-dash/assets/5731733/87dced8b-632b-4099-9429-81552fd5efad">

For the benefits, it's hard to say without being able to build these apps. In next-article, version 1 of date-fns seems to be responsible for a large proportion of the client side JS:
<img width="1470" alt="iundle js" src="https://github.com/Financial-Times/x-dash/assets/5731733/9b9e1da5-ba5f-4694-9838-e2fb0891b78a">

Adding tests to increase reliability and confidence is something I'm happy to do, so please let me know if there's any more I can add :)